### PR TITLE
Add coodlown_bypass decorators and cog local.

### PR DESCRIPTION
**Pull request by EvieePy**
### Summary

Allow for cooldowns to be bypassed by several methods:

`@bot.cooldown_bypass` (Applies to all commands)
`@command.cooldown_bypass` (Applies to a single command)
`cog_cooldown_bypass` (Applies to all cog commands)

These checks allow the user to determine whether or not a cooldown(s) should be bypassed.

If the check returns True, the cooldown will be bypassed and no ratelimits will apply, else the cooldown will run as normal.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by CapnS** [comment]
My personal opinion is that the check should go the other way e.g. the cooldown gets run if the check returns True. This would keep in line with the other checks in this library where the action happens if the check returns True.

File: discord/ext/commands/bot.py [hunk 1]
```diff
@@ -362,6 +363,48 @@ def after_invoke(self, coro):
...
+            async def my_command(ctx):
+                await ctx.send('This message can be sent unlimitedly by the guild owner, due to my cooldown bypass.)
+
+        Raises
+        -------
+        discord.ClientException
+            The coroutine is not actually a coroutine.
+        """
+        if not asyncio.iscoroutinefunction(coro):
+            raise discord.ClientException('The cooldown bypass must be a coroutine.')
```
Comment (Rapptz) [inline discord/ext/commands/bot.py]:
```suggestion
            raise TypeError('The cooldown bypass must be a coroutine.')
```

Same everywhere else.

File: discord/ext/commands/bot.py [hunk 2]
```diff
@@ -362,6 +363,48 @@ def after_invoke(self, coro):
...
+                return False
+
+            @bot.command()
+            @commands.cooldown(1, 30, type=commands.BucketType.guild)
+            async def my_command(ctx):
+                await ctx.send('This message can be sent unlimitedly by the guild owner, due to my cooldown bypass.)
+
+        Raises
+        -------
+        discord.ClientException
```
Comment (Rapptz) [inline discord/ext/commands/bot.py]:
```suggestion
        TypeError
```

Same everywhere else.

File: discord/ext/commands/bot.py [hunk 3]
```diff
@@ -362,6 +363,48 @@ def after_invoke(self, coro):
...
+            The coroutine is not actually a coroutine.
```
Comment (Rapptz) [inline discord/ext/commands/bot.py]:
```suggestion
            The coroutine passed is not actually a coroutine.
```

Same everywhere else.

**Comment by scragly** [comment]
Since we don't want the cooldown check to be primarily identified with bypassing in it's functionality and naming, we can just no call it a bypass and treat it the same as every other predicated check in the library where returning True applies the thing it's checking, in this case the cooldown.

Effectively, this flips the current True/False behaviour, and allows us to adjust the name to something primarily about it being the predicate check of the cooldown.

**Pull request closed without merging**